kernel/network/network_io.c

```diff
@@ -94,9 +108,12 @@ ntoh_mac(mac_address_t mac)
     }
 }
 
-// Check whether current running mode is set to little or big endian
-uint8_t
+/*
+ * Check whether current running mode is set to little or big endian
+ * via looking at the system register.
+ */
+bool
 is_big_endian()
 {
-  return read32(SYS_CFGDATA2) & (1 << 1);
+  return read32(SYS_CFGDATA2) & (1 << 1) > 0;
```

Comment:
I would cast it to a bool (if that is possible) because it better reflects the semantic of the code